packages/store/addon/-private/system/fetch-manager.ts

```diff
@@ -0,0 +1,554 @@
+import { RecordIdentifier, identifierForIM } from "./record-identifier";
+import { default as RSVP, Promise } from 'rsvp';
+import { DEBUG } from '@glimmer/env';
+import { run as emberRunLoop } from '@ember/runloop';
+import Adapter from "@ember/test/adapter";
+import { AdapterCache } from "./adapter-cache";
+import { assert, warn, inspect } from '@ember/debug';
+import Snapshot from './snapshot';
+import { guardDestroyedStore, _guard, _bind, _objectIsAlive } from './store/common';
+import { normalizeResponseHelper } from './store/serializer-response';
+import { serializerForAdapter } from './store/serializers';
+import { InvalidError } from '@ember-data/adapter/error';
+import coerceId from './coerce-id';
+import Ember from 'ember';
+
+import {
+  _findHasMany,
+  _findBelongsTo,
+  _findAll,
+  _query,
+  _queryRecord,
+} from './store/finders';
+import RequestCache from "./request-cache";
+import { RecordData, RelationshipRecordData } from "./model/record-data";
+
+function payloadIsNotBlank(adapterPayload) {
```

Comment:
as this currently exists, it could return `boolean | number` which is kind of strange